presto-array/src/main/java/com/facebook/presto/array/LongBigArray.java

```diff
@@ -125,7 +147,7 @@ private void grow(long length)
 
         // grow base array if necessary
         if (array.length < requiredSegments) {
-            array = Arrays.copyOf(array, requiredSegments);
+            array = Arrays.copyOf(array, toIntExact(round(requiredSegments * 1.5)));
```

Comment:
Typical array expanding is based on existing size `Math.max(toIntExact(round(array.length * 1.5)), requiredSegments)`